subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -716,6 +813,86 @@ static void pe_drs_send_swap_run(void *obj)
 	}
 }
 
+/**
+ * PE_Get_Sink_Cap Entry state
+ * @brief 8.3.3.18.7.1 PE_DR_SRC_Get_Source_Cap State
+ * @brief 8.3.3.18.9.1 PE_DR_SNK_Get_Sink_Cap State
+ */
+void pe_get_sink_cap_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * On entry to the PE_DR_SNK_Get_Sink_Cap state the Policy Engine
+	 * Shall send a Get_Sink_Cap Message and initialize and run the
+	 * SenderResponseTimer.
+	 */
+	pe_send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_DR_SWAP);
+	/* Initialize Submachine */
+	pe->submachine = 0;
+}
+
+/**
+ * @brief PE_Get_Sink_Cap Run state
+ *        NOTE: Sender Response Timer is handled in super state.
+ */
+void pe_get_sink_cap_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	struct usbc_port_data *data = dev->data;
+	struct protocol_layer_rx_t *prl_rx = data->prl_rx;
+	union pd_header header;
+
+	switch (pe->submachine) {
+	case 0:
+		if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+			pe->submachine++;
+		}
+		break;
+	case 1:
+		if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_MSG_RECEIVED)) {
+			header = prl_rx->emsg.header;
+
+			if (prl_rx->emsg.type == PD_PACKET_SOP) {
+				if (received_data_message(dev, header, PD_DATA_SINK_CAP)) {
+#ifdef CONFIG_USBC_CSM_SOURCE_ONLY
+					uint32_t *pdos = (uint32_t *)prl_rx->emsg.data;
+					uint32_t num_pdos =
+					PD_CONVERT_BYTES_TO_PD_HEADER_COUNT(prl_rx->emsg.len);
+
+					policy_set_port_partner_snk_cap(dev, pdos, num_pdos);
+					pe_set_state(dev, PE_SRC_READY);
+#else
+					pe_set_state(dev, PE_SNK_READY);
+#endif
+				} else if (received_control_message(dev, header, PD_CTRL_REJECT) ||
+					received_control_message(dev,
+							header, PD_CTRL_NOT_SUPPORTED)) {
+#ifdef CONFIG_USBC_CSM_SOURCE_ONLY
+					pe_set_state(dev, PE_SRC_READY);
+#else
+					pe_set_state(dev, PE_SNK_READY);
+#endif
+				}
+				/* Unexpected messages fall through to soft reset */
```

Comment:
Without some `return`s above, all messages will fall through to soft reset.

Reply:
good catch. fixed